### ISSUE TITLE
 Memory and connection leaks when using custom add rabbitmq health check

### ISSUE THREAD

**Issue by virtualcca** [Xabaril/AspNetCore.Diagnostics.HealthChecks#578]
**What happened**:
when i use custome add rabbitmq healthcheck, like this:
```
 services.AddHealthChecks().Add(new HealthCheckRegistration("✉️RabbitMQ", sp =>
                    new RabbitMQHealthCheck(new Uri("amqp://guest:guest@ip:port"), null), null, null));
```
then constantly call the health check interface
will constantly create rabbitmq connection instances

The following memory comparison is that after 5 calls, you can see that there are 5 new instances of rabbitmq
![image](https://user-images.githubusercontent.com/5656078/87038372-34468400-c220-11ea-86fb-3b2dcfb382c3.png)

**How to reproduce it (as minimally and precisely as possible)**:
Use the following code to initialize an aspnetcore health check, and then continue to call

**Source code sample**:
in startup.cs
```
    public void ConfigureServices(IServiceCollection services)
        {
            //other
            services.AddHealthChecks().Add(new HealthCheckRegistration("✉️RabbitMQ", sp =>
                    new RabbitMQHealthCheck(new Uri("amqp://guest:guest@ip:port"), null), null, null));
        }
```

- .NET Core version Net Core 3.1.5
- Healthchecks version AspNetCore.HealthChecks.Rabbitmq 3.1.2
- Operative system: Win10 1904
- Others:

**Comment by unaizorrilla** [comment]
Hi @virtualcca 

An extension method is available to properly use the registration and avoid create multiple connections:

Also, you can register the IConnectionFactory directly (instead of IConnection)

```
services
        .AddSingleton<IConnection>(sp=>
        {
            var factory = new ConnectionFactory()
            {
                Uri = new Uri("amqps://user:pass@host/vhost"),
                AutomaticRecoveryEnabled = true
            };

            return  factory.CreateConnection();
        })
        .AddHealthChecks()
        .AddRabbitMQ();
```

**Comment by virtualcca** [comment]
> Hi @virtualcca
> 
> An extension method is available to properly use the registration and avoid create multiple connections:
> 
> Also, you can register the IConnectionFactory directly (instead of IConnection)
> 
> ```
> services
>         .AddSingleton<IConnection>(sp=>
>         {
>             var factory = new ConnectionFactory()
>             {
>                 Uri = new Uri("amqps://user:pass@host/vhost"),
>                 AutomaticRecoveryEnabled = true
>             };
> 
>             return  factory.CreateConnection();
>         })
>         .AddHealthChecks()
>         .AddRabbitMQ();
> ```


I can know if I use it directly like
```
services.AddHealthChecks().AddRabbitMQ(rabbitConnectionString: "amqp://guest:guest@ip:port");
```
is ok

But I hope that the problems added by the custom can also be solved, or maybe this overload can be specially marked under the possible problems

After all, an overload is exposed, then it is either functioning normally, if there are other prerequisites, it should also be paid attention to in some form, otherwise it is too easy to misuse

**Comment by unaizorrilla** [comment]
We have comments on the RabbitMQ readme

 https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/tree/master/src/HealthChecks.RabbitMQ 

but we can add this into Remarks comments on code!

Thanks!

Can I close the issue?

**Comment by virtualcca** [comment]
> We have comments on the RabbitMQ readme
> 
> https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/tree/master/src/HealthChecks.RabbitMQ
> 
> but we can add this into Remarks comments on code!
> 
> Thanks!
> 
> Can I close the issue?

Ok thanks for your answer

**Comment by unaizorrilla** [comment]
Closed with #587
